### ISSUE TITLE
Feature requests

### ISSUE THREAD

**Issue by valinet** [valinet/ExplorerPatcher#13]
Please comment here with any feature request, and we will discuss it and see what we can do. Use this so that the issues panel does not get cluttered with things that may never end up being 'solved'.

Thank you.

**Comment by Gaurav-Original-ClassicShellTester** [comment]
Feature request:

Switch to balloon notifications since the toasts accumulate in Notification Center (Win+N) and have to be cleared every time ExplorerPatcher is updated. Otherwise multiple toast notifications accumulate as Explorer restarts many times. As far as I know, balloon tips show only when user is active (mouse is moving) otherwise they are queued if system is considered idle/user is away. Like toasts, they disappear on their own fading out but without requiring further action to clear them from any Action Center/Notification Center.

They can be enabled with HKEY_CURRENT_USER\SOFTWARE\Policies\Microsoft\Windows\Explorer
EnableLegacyBalloonNotifications=1

But apps have to specifically use them, I think? I don't know if apps can use them without EnableLegacyBalloonNotifications set to 1.

**Comment by valinet** [comment]
You mean to switch the notifications from ExplorerPatcher from toasts to baloons or the notifications in general, for the whole system?

**Comment by Gaurav-Original-ClassicShellTester** [comment]
I mean for ExplorerPatcher, not the whole system. Whole system the user can switch on their own using the Registry tweak. But since ExplorerPatcher shows them every time Explorer is restarted until all the symbols are downloaded, they accumulate and get annoying, have to be cleared from Notification Center before the next arrived notification can be read.

**Comment by valinet** [comment]
Yeah, I don't think it is an issue, the application does not have to download symbols so frequently. Maybe in the future I will take that functionality out altogether and just hard code offsets for each new build. Anyway, I prefer toasts because:

* they look nicer =)))
* I don't have to carry an icon with the app; for balloons, you need to put an icon in the system tray; for toasts, you specify the app ID of the app you want the toast to come from, in my case `Microsoft.Windows.Explorer` and the system takes care of displaying the proper icon (apps registered for toasts can be obtained in PowerShell using `Get-StartApps`)
* most importantly, on default Windows configurations, balloons are automatically transformed in toasts anyway, unless you employ the registry tweak you suggested, so it would achieve the same thing but via a different API that does not allow as much customization to the toast as the toasts API itself; I specifically just tested `Shell_NotifyIcon`: with default settings, Windows still makes and displays a toast out of the balloon data and it also persists in the action center despite having no action associated with it
* these toasts are anyway just a gimmick; I did not spend time making them work for this project, I just had the code for them from a previous project where I used them ([ThunderbirdToasts](https://github.com/valinet/ThunderbirdToasts); I could have taken them out as well; or, make them standard message boxes that you click OK and then the app moves on with its job; I preferred toasts because they are asynchronous and the app/user can continue doing its job

Considering this, I have to admit that the fact that the "x" on a notification does not dismiss it, but rather it moves it to the notification center is counter intuitive and kind of stupid, in my opinion. There should have been 2 buttons, one to dismiss it and one to get rid of it for good, or have the swipe move it to the notification center and the "x" close it, idk.

Anyway, I'd consider a good compromise to be if the app still showed toasts, but the toasts would simply go away, without remaining in the notification center. I am not sure whether that is doable or not with the API Windows offers. If you could look it up, find some answer regarding this, that'd be great. You can play with toasts using this app: [Toast](https://github.com/go-toast/toast). Or, I could have another option in the `settings.ini` file where you configure the use of `MessageBox`es instead of toasts for these messages.

But yeah, I don't know, I don't feel like this is that important. They are anyway grouped in the notification center, you just click once with the middle button to dismiss them for good.

**Comment by Gaurav-Original-ClassicShellTester** [comment]
Oh ok that's alright then as it is :) Windows 11 is still evolving. Maybe they will fix the toasts so that if they are actually swiped away, they don't get stored in the Notification center.

**Comment by hui99995** [comment]
![image](https://user-images.githubusercontent.com/14900130/132534716-a8bc51ac-2059-4a37-8a0a-2a76f5d06f11.png)
目前的版本改小任务栏日期会被遮挡住，这一块存在下移，能上移一部分不？

In the current version, the date of the taskbar will be blocked. This piece can be moved down. Can you move up a part?

**Comment by valinet** [comment]
That's the new taskbar, this app works with the classic taskbar only. The new taskbar is intended to be used with the customization options that Windows provides. `TaskbarSi` is not an option offered officially, so they probably do not test for it, and thus using it breaks the new taskbar. All in all, the new taskbar only has one size option: the default one. Personally, I don't know how the new taskbar works or what it is made of, so no, this software cannot provide help with that, sorry.

**Comment by valinet** [comment]
> 
> 
> ![image](https://user-images.githubusercontent.com/14900130/132534716-a8bc51ac-2059-4a37-8a0a-2a76f5d06f11.png)
> 目前的版本改小任务栏日期会被遮挡住，这一块存在下移，能上移一部分不？
> 
> In the current version, the date of the taskbar will be blocked. This piece can be moved down. Can you move up a part?

On the other hand, this just got released, maybe check out the features it offers and whether it can help you: http://startisback.com/tbd/

**Comment by hui99995** [comment]
Yes, thank you

**Comment by duk6046** [comment]
Suggests taskbar center alignment function like this.
![2021-10-02 213350](https://user-images.githubusercontent.com/32978164/135716368-974b82d8-5dee-4f88-afd2-70df61354e9f.png)

**Comment by valinet** [comment]
> 
> 
> Suggests taskbar center alignment function like this. ![2021-10-02 213350](https://user-images.githubusercontent.com/32978164/135716368-974b82d8-5dee-4f88-afd2-70df61354e9f.png)

Idk why TaskbarX does not work with the main taskbar, I have tried it. I'd prefer a third party solution, idk... StartIsBack has that, afaik. Aren't there any other tools, besides TaskbarX and StartIsBack, that offer this? Or maybe fix TaskbarX...?

**Comment by valinet** [comment]
@duk6046 Okay, I have a solution: please use [TaskbarX](https://github.com/ChrisAnd1998/TaskbarX). It is a wonderful program that does just this, it has a couple of features on top, like animations, hide Start button etc, and it works really well. It has a slight problem in Windows 11, where it won't center the main taskbar. I already forked the program and fixed that, so, in the mean time, before my PR gets accepted into the upstream and a new release is made, I suggest you do either the following:
[TaskbarX.zip](https://github.com/valinet/ExplorerPatcher/files/7271518/TaskbarX.zip)

* Clone my fork and compile "TaskbarX.exe" in there and replace the original one from the Github release with this one
* I attach a ZIP with the "TaskbarX.exe" that I have compiled; use that in the mean time or in place of the file from Github

This is a temporary fix. Hopefully, the proper fix will be merged into upstream soon and all of this won't be required.
Thanks

[TaskbarX.zip](https://github.com/valinet/ExplorerPatcher/files/7271525/TaskbarX.zip)
.

**Comment by duk6046** [comment]
Thank you so much

**Comment by valinet** [comment]
Yeah, I also updated the code to have the Start button follow the buttons. it is available in my fork. Here is another precompiled version with this change: [TaskbarX.zip](https://github.com/valinet/ExplorerPatcher/files/7271765/TaskbarX.zip)

**Comment by Gaurav-Original-ClassicShellTester** [comment]
In the latest release, "The system tray network icon now opens the WiFi list in Control Center". This is awesome. Thank you!

Feature request: Can clicking it again close it while it is open? That's how the nicer Windows 7 or Windows 10 Network flyouts behave :) Even better, the Bluetooth and USB Safely remove icons could also close if clicked while they are open? The Volume, Power, Clock/Date, Language and Action Center always have this behavior.

**Comment by valinet** [comment]
> 
> 
> In the latest release, "The system tray network icon now opens the WiFi list in Control Center". This is awesome. Thank you!
> 
> Feature request: Can clicking it again close it while it is open? That's how the nicer Windows 7 or Windows 10 Network flyouts behave :) Even better, the Bluetooth and USB Safely remove icons could also close if clicked while they are open? The Volume, Power, Clock/Date, Language and Action Center always have this behavior.

The part with toggling the menus on Bluetooth and Safe to Remove, that is doable as well. It just takes time, I am all alone. Anyway, when I touch it, I'll have the menus open centered relative to the icon, same as the other flyouts work.

**Comment by valinet** [comment]
> 
> 
> @valinet
> 
> Feature Request
> 
> It would be nice to have [Win11DisableRoundedCorners](https://github.com/valinet/Win11DisableRoundedCorners) included in the Win+X > Properties GUI as an option.

Yeah, I could do that eventually...

**Comment by valinet** [comment]
> 
> 
> > Great, that's nice to hear. You know what would be awesome? Clicking the search box to open the Everything search. Maybe even search in the window, but that's not a must, initially at least. That would be nuts. As we have learned, we just have to implement a COM component with the same structure as `{64bc32b5-4eec-4de7-972d-bd8bd0324537}` and have Explorer use that instead of its crappy attempts via the `TreatAs` mechanism.
> 
> Well right now I set hotkey_explorer_path_search=1 (from default 0) in Everything.ini. When this option is enabled, pressing the Everything New window hotkey from an Explorer window will set the search to the current path in Windows Explorer and we can just begin typing to filter/find items from that path.
> 
> To open Everything from anywhere without this behavior, simply set a different Show Window hotkey or Toggle Window hotkey. Although I don't know how to make a click in the Explorer search box, send that New window hotkey which I have assigned in Everything. :D

![evgif2](https://user-images.githubusercontent.com/6503598/136576873-76ec0c65-4d6c-4d3a-aef4-475e2b0530f0.gif)

Working proof of concept code I played with yesterday. Again, should mature and be available in a release at some point. Basically, one clicks or focuses the Explorer search bar and then it starts typing in Everything. Quite a few things have to be ironed out, but it has potential. Should work in Open/Save file dialogs as well.

**Comment by valinet** [comment]
I just pushed a new release containing the various minor fixes and quality of life improvements discussed though this thread (some are yet to be pushed to `master`). This is the last release for a while. The application will go through a reorganization and refactoring process in order to better support new and enhanced capabilities coming soon. Updates will continue as commits to the repo, but new releases will resume only after the bulk of this work has been completed. Thank you!

**Comment by Gaurav-Original-ClassicShellTester** [comment]
Well, the menu fixes and network icon in 22000.194.24 work great - all 3 close upon 2nd click. When both EP and SIB are enabled, only the Bluetooth icon closes upon 2nd click, the USB Safely Remove doesn't - not an issue for me - just thought I would mention it.

The Explorer search box still reverts to classic when both ExplorerPatcher and StartAllBack are enabled (StartAllBack mainly I need for skinning taskbar buttons, otherwise I wouldn't need it - since it can apply msstyles from 7/8.1 to make the horrible flat buttons without any separation lines look better). But I can live with it - since I prefer the other ExplorerPatcher enhancements anyway like Win+X, Win+C.

No requirement now for the DLL to even be in 2 locations is GREAT! Makes the project very easy to install and upgrade, especially since Windows 11 major build upgrades only migrated the DLL in %windir% but the others went to C:\Windows.old.

**Comment by Anix-x** [comment]
Under Windows Classic theme the context menus in File Explorer are missing text
![image](https://user-images.githubusercontent.com/92230793/136674338-2be5f318-3bdd-4cbe-8de6-046de39cd234.png)

This is a known issue since certain builds of Windows 10. The workaround is to use ExplorerContextMenuTweaker program 
( https://github.com/rikka0w0/ExplorerContextMenuTweaker ).

The program is buggy though: does not work the first time the context meny is called and does not affect submenus except the first one.

But since ExplorerPatcher already implements a similar functionality of another program TaskbarContextMenuTweaker by the same author, it would be great and logical if this functionalty was also implemented. The currently-implemented non-immersive context menus by ExplorerPatcher are still empty under Windows Classic theme.

**Comment by valinet** [comment]
> 
> 
> Under Windows Classic theme the context menus in File Explorer are missing text ![image](https://user-images.githubusercontent.com/92230793/136674338-2be5f318-3bdd-4cbe-8de6-046de39cd234.png)
> 
> This is a known issue since certain builds of Windows 10. The workaround is to use ExplorerContextMenuTweaker program ( https://github.com/rikka0w0/ExplorerContextMenuTweaker ).
> 
> The program is buggy though: does not work the first time the context meny is called and does not affect submenus except the first one.
> 
> But since ExplorerPatcher already implements a similar functionality of another program TaskbarContextMenuTweaker by the same author, it would be great and logical if this functionalty was also implemented. The currently-implemented non-immersive context menus by ExplorerPatcher are still empty under Windows Classic theme.

Reply: https://github.com/valinet/ExplorerPatcher/issues/49

**Comment by valinet** [comment]
@Adiker It may come in a future version, I have it on the to do list at some point, but right now the project will go through some internal cleanup and reorganization first, to better support new and enhanced capabilities. Thanks.

**Comment by valinet** [comment]
The place where the Start menu is opened is controlled by the taskbar alignment setting Left/Center. It’s in the taskbar section of Settings and also in the Properties GUI.

Regarding the Start menu icon, it’s on my to do list, I don’t know how to do it or when it is going to be ready at the moment, but the plan is to implement that eventually. As well, I have to do some housekeeping first before adding new capabilities to the project.

Thanks.

**Comment by rti7743** [comment]
Is it possible to hide of the CommandBar in File Explorer?

I do not want the CommandBar to appear as it does in Windows XP.
I would like to see a mode where CommandBar is not displayed like in Windows XP, not in Windows 10 or Windows 7.

In Windows 7 and Windows 10, there is a way to edit and erase Shellstyle.dll in Resource Hacker, but it would be useful if this software could erase it more easily.
Since you have already created an implementation that turns off the search bar, could you create feature of that routine?

How to Auto-Hide or Disable Command Bar in Windows Vista and 7 Explorer?
https://www.askvg.com/how-to-make-folder-band-auto-hidden-in-windows-vista/

How to remove the ribbon and command bar in Windows 10
https://superuser.com/questions/1317370/how-to-remove-the-ribbon-and-command-bar-in-windows-10

**Comment by RicMendieta** [comment]
The space between icons on the taskbar can be reduce?  That way icons on the taskbar can be compact and more can fit on the same space.
